src/algebra/group_power.lean

```diff
@@ -142,6 +142,16 @@ by rw [mul_comm, pow_mul]
 theorem mul_nsmul (a : A) (m n : ℕ) : m * n •ℕ a = m •ℕ (n •ℕ a) :=
 @pow_mul' (multiplicative A) _ a m n
 
+theorem pow_mul_pow_sub (a : M) {m n : ℕ} (h : m ≤ n) : a ^ m * a ^ (n - m) = a ^ n :=
+by rw [←pow_add, nat.add_sub_cancel' h]
+theorem nsmul_add_sub_nsmul (a : A) {m n : ℕ} (h : m ≤ n) : (m •ℕ a) + ((n - m) •ℕ a) = n •ℕ a :=
```

Comment:
There are quite a few lemmas in that file formatted this way already. It's because they always come in pairs. Should I really add the newlines everywhere?